include/xsimd/arch/generic/xsimd_generic_math.hpp

```diff
@@ -962,6 +962,21 @@ namespace xsimd
             return { rem1 - 2. * re * si * si, re * isin };
         }
 
+        // polar
+        template <class A, class T>
+        inline batch<std::complex<T>, A> polar(const batch<T, A>& r, const batch<T, A>& theta, requires_arch<generic>) noexcept
+        {
+            auto sincosTheta = sincos(theta);
+            return { r * sincosTheta.second, r * sincosTheta.first };
+        }
+
+        template <class A, class T>
+        inline batch<std::complex<T>, A> polar(const batch<T, A>& theta, requires_arch<generic>) noexcept
```

Comment:
shouldn't this be `r` instead of `theta`?

Reply:
And then the formula is different, of course. But that would match the C++ api, as https://en.cppreference.com/w/cpp/numeric/complex/polar